include/wil/registry.h

```diff
@@ -17,16 +17,2971 @@
 
 #include <winreg.h>
 #include <new.h> // new(std::nothrow)
-#include "resource.h" // unique_hkey
+#include "registry_helpers.h"
+#include "resource.h"
+
+#ifndef __WIL_WINREG_
+#error __WIL_WINREG_ is required for ::wil::unique_hkey support; ensure the raw registry APIs (like RegCreateKeyExA) are available

```

Comment:
This makes me a little uneasy. `registry.h` is an existing header file, so this could possibly be breaking such consumers. On top of that, if I take a look at the conditions under which `__WIL_WINREG_` get defined, it will be impossible for some consumers to satisfy this constraint
```c++
#if defined(_WINREG_) && !defined(__WIL_WINREG_) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && !defined(WIL_KERNEL_MODE)
```
In particular, anything targeting the APP/GAMES/etc. partitions will break. IMO it makes more sense to conditionally define the new features to depend on the presence of `__WIL_WINREG_` (which AFAICT won't cause issues because the file includes `winreg.h` before `resource.h`)

Reply:
_Aside: The restriction on the `DESKTOP` partition seems overly constrained and we should probably fix that_

Reply:
Thanks. The existing registry.h code directly references unique_hkey, so I'll do the same and remove this ifndef block completely.